cmd/mayactl/app/command/volume_info.go

```diff
@@ -0,0 +1,102 @@
+	"github.com/openebs/maya/types/v1"
+	"github.com/spf13/cobra"
+)
+
+var (
+	volumeInfoCommandHelpText = `
+	    Usage: mayactl volume info --volname <vol>
+
+            This command queries the info of the volume.
+        `
+)
+
+// CmdVolumeAccessOptions is used to store the value of flags used in the cli
+type CmdVolumeInfoOptions struct {
+	volName string
+}
+
+// NewCmdVolumeAccessMode shows info of OpenEBS Volume
+func NewCmdVolumeInfo() *cobra.Command {
+	options := CmdVolumeInfoOptions{}
+	cmd := &cobra.Command{
+		Use:     "info",
+		Short:   "Displays the info of Volume",
+		Long:    volumeInfoCommandHelpText,
+		Example: `mayactl volume info --volname vol`,
+		Run: func(cmd *cobra.Command, args []string) {
+			util.CheckErr(options.Validate(cmd), util.Fatal)
+			util.CheckErr(options.RunVolumeInfo(cmd), util.Fatal)
+		},
+	}
+	cmd.Flags().StringVarP(&options.volName, "volname", "n", options.volName,
+		"unique volume name.")
+	return cmd
+}
+
+// Validate verifies the command whether volName is passed or not.
+func (c *CmdVolumeInfoOptions) Validate(cmd *cobra.Command) error {
+	if c.volName == "" {
+		return errors.New("--volname is missing. Please specify a volume name created")
+	}
+	return nil
+}
+
+// TODO : Add more volume information
+// RunVolumeAccessMode runs info command and display the outputs in standard I/O.
+// Currently It displays volume access mode only.
+func (c *CmdVolumeInfoOptions) RunVolumeInfo(cmd *cobra.Command) error {
+	annotation := &Annotations{}
+	// GetVolumeAnnotation is called to get the volume controller's info such as
+	// controller's IP, status, iqn, replica IPs etc.
+	err := annotation.GetVolAnnotations(c.volName)
+	if err != nil {
+		fmt.Println("Can't get annotation, found error ", err)
+		return nil
+	}
+	if err != nil || annotation == nil {
+		fmt.Println(err)
+		return nil
+	}
+	if annotation.ControllerStatus != "Running" {
+		fmt.Println("Volume not reachable")
+		return nil
+	}
+	// Initiallize an instance of ReplicaCollection, json response recieved from the
+	// controllerIP:9501/v1/replicas is to be parsed into this structure via GetVolumeAccessMode.
+	// GetVolumeAccessMode only gives the details related to the access modes.
+	collection := client.ReplicaCollection{}
+	controllerClient := client.ControllerClient{}
+	_, err = controllerClient.GetVolumeStats(annotation.ClusterIP+v1.ControllerPort, v1.InfoAPI, &collection)
+	if err != nil {
+		fmt.Println("Could not decode the json response", err)
+		return err
+	}
+	// Printing the o/p in tabular format.
+	q := tabwriter.NewWriter(os.Stdout, v1.MinWidth, v1.MaxWidth, v1.Padding, ' ', tabwriter.AlignRight|tabwriter.Debug)
+	fmt.Fprintf(q, "ReplicaIP\tMode\t\n")
+	fmt.Fprintf(q, "\t\t\n")
+	for key, _ := range collection.Data {
+		address := strings.TrimPrefix(collection.Data[key].Address, "tcp://")
+		address = strings.TrimSuffix(address, ":9502")
+		fmt.Fprintf(q, "%s\t%s\t\n", address, collection.Data[key].Mode)
```

Comment:
logic to display & sanitation logic are mixed. It might help to keep them separate. Need to make some of these pieces reusable than repeating it across commands.